.github/workflows/build-and-test.yml

```diff
@@ -4,7 +4,13 @@ on:
     branches: [ main ]
     tags:
       - 'v[0-9]+.[0-9]+.[0-9]+*'
+    paths-ignore:
+      - '.github/**'
```

Comment:
how would this work if someone is making a change to build-and-test that needs to be run in order to see if it works?